extensions/script/ExecuteScript.h

```diff
@@ -28,17 +28,95 @@
 
 #include "ScriptEngine.h"
 #include "ScriptProcessContext.h"
+#include "utils/Enum.h"
 
 namespace org {
 namespace apache {
 namespace nifi {
 namespace minifi {
+
+#ifdef PYTHON_SUPPORT
+namespace python {
+class PythonScriptEngine;
+}
+#endif  // PYTHON_SUPPORT
+
 namespace processors {
 
+class ScriptEngineFactory {
+ public:
+  ScriptEngineFactory(core::Relationship& success, core::Relationship& failure, std::shared_ptr<core::logging::Logger> logger);
+
+  template<typename T>
+  std::shared_ptr<T> createEngine() const {
+    auto engine = std::make_shared<T>();
```

Comment:
I think it would make sense to add some restriction, e.g. that `T` is derived from `ScriptEngine`.
```suggestion
  template<typename T>
  std::enable_if_t<std::is_base_of_v<script::ScriptEngine, T>, std::shared_ptr<T>> createEngine() const {
    auto engine = std::make_shared<T>();
```

Reply:
Good idea, added restriction in e1583ef65af13edcd0d5b07da6eee5ea84ba3cbd